concordium-consensus/src/Concordium/KonsensusV1/TreeState/Types.hs

```diff
@@ -203,6 +203,44 @@ data RoundStatus = RoundStatus
     }
     deriving (Show, Eq)
 
+-- |Advance the provided 'RoundStatus' to the provided 'Round'.
+-- If @Maybe (TimeoutCertificate, QuorumCertificate)@ is 'Nothing'
+-- then the round advances as it were a decendant of 'Round' where a 'QuorumCertificate' was successfully
+-- produced. Otherwise if @Just ..@ then it means that the old round timed out.
+-- All properties from the old 'RoundStatus' are being carried over to new 'RoundStatus'
+-- except the following.
+-- * 'rsCurrentRound' will become the provided 'Round'.
+-- * 'rsCurrentQuorumSignatureMessages' will be 'emptySignatureMessages'.
+-- * 'rsCurrentTimeoutSignatureMessages' will be 'emptySignatureMessages'.
+-- * 'rsPreviousRoundTC' will be either 'Absent' (if 'Nothing'
+advanceRoundStatus ::
+    -- |The round to advance to.
+    Round ->
+    -- |'Nothing' if advancing from a round
+    -- which successfully produced a 'QuorumCertificate'.
+    -- @Just (..,..)@ if advancing from a round which produced a
+    -- 'TimeoutCertificate'.
+    -- In the latter case, then 'TimeoutCertificate' provided must be
+    -- the 'TimeoutCertificate' for the round that timed out and the 'QuorumCertificate'
+    -- is the one verifing the 'TimeoutCertificate'.
+    Maybe (TimeoutCertificate, QuorumCertificate) ->
+    -- |The 'RoundStatus' we are advancing from.
+    RoundStatus ->
+    -- |The advanced 'RoundStatus'.
+    RoundStatus
+advanceRoundStatus toRound mTcQc RoundStatus{..} =
```

Comment:
This function simply updates fields on a record, so I don't think there's any benefit to having it separated from `advanceRound` like this.

Reply:
It's gonna do more when it's merged with: https://github.com/Concordium/concordium-node/pull/787

In particular it's also gonna clear up the signature messages for the current round.